SIPS/sip-23.md

```diff
@@ -0,0 +1,627 @@
+  onSubmit?: ((formData: Record<string, string>) => void) | undefined;
+};
+```
+
+##### Example
+
+```typescript jsx
+<Form onSubmit={handleSubmit}>
+  <Field label="Name">
+    <Input type="text" />
+  </Field>
+</Form>
+```
+
+#### Heading
+
+The `Heading` component renders a heading element. It accepts a `children` prop
+which can be a `string` or an array of `string`s.
+
+```typescript
+type HeadingProps = {
+  children: string | string[];
+};
+
+const Heading: SnapComponent<HeadingProps>;
+```
+
+##### Example
+
+```typescript jsx
+<Heading>My Snap</Heading>
+```
+
+#### Image
+
+The `Image` component renders an image element. It accepts a `src` prop which is
+a string representing the image source. The component also accepts an optional
+`alt` prop which is a string representing the image description.
+
+The `src` MUST be an SVG string. Other image formats are not supported, but
+PNG and JPEG images MAY be used inside the SVG as data URIs.
+
+```typescript
+type ImageProps = {
+  src: string;
+  alt?: string | undefined;
+};
+
+const Image: SnapComponent<ImageProps>;
+```
+
+##### Example
+
+```typescript jsx
+<Image src="<svg>...</svg>" alt="My image" />
+```
+
+#### Input
+
+The `Input` component renders an input element. It accepts a `name` prop which
+is a string representing the input name, a `type` prop which is a string
+representing the input type, and an optional `value` prop which is a string
+representing the input value. The component also accepts an optional
+`placeholder` prop which is a string representing the input placeholder, and an
+optional `onChange` prop which is a function to be called when the input value
+changes.
+
+The `type` MUST be one of `'text'`, `'password'`, or `'number'`.
+
+```typescript
+type InputProps = {
+  name: string;
+  type: 'text' | 'password' | 'number';
+  value?: string | undefined;
+  placeholder?: string | undefined;
+  onChange?: ((value: string) => void) | undefined;
+};
+
+const Input: SnapComponent<InputProps>;
+```
+
+##### Example
+
+```typescript jsx
+<Input name="name" type="text" placeholder="Enter your name" onChange={handleChange} />
+```
+
+#### Italic
+
+The `Italic` component renders a text element in italic. It accepts a `children`
+prop which can be a `string` or an array of `string`s. The component MUST be
+used inside a `Text` component.
+
+```typescript
+type ItalicProps = {
+  children: string | string[];
+};
+
+const Italic: SnapComponent<ItalicProps>;
+```
+
+##### Example
+
+```typescript jsx
+<Text>
+  Hello, <Italic>world</Italic>!
+</Text>
+```
+
+#### Link
+
+The `Link` component renders a link element. It accepts a `children` prop which
+can be a `string` or an array of `string`s, and an `href` prop which is a string
+representing the link URL.
+
+The `href` MUST be an absolute URL. Relative URLs are not supported. It MUST
+start with `https://` or `mailto:`.
+
+```typescript
+type LinkProps = {
+  children: string | string[];
```

Comment:
What does providing a string array as the children look like?

Reply:
```tsx
<Link>
  Some text with a {variable}
</Link>
```

basically gets turned into

```tsx
Link([
  'Some text with a',
  variable
]);
```